framework/include/reporters/ReporterState.h

```diff
@@ -190,19 +197,36 @@ ReporterState<T>::copyValuesBack()
     (*iter) = (*std::next(iter));
 }
 
+template <typename T>
+void
+ReporterState<T>::storeInternal(std::ostream & stream)
+{
+  // Store the container size
+  std::size_t size = this->get().size();
+  dataStore(stream, size, nullptr);
+
+  // Store each entry of the list directly into the storage
+  for (auto & val : this->set())
+    storeHelper(stream, val, nullptr);
+}
+
 template <typename T>
 void
 ReporterState<T>::loadInternal(std::istream & stream)
 {
   // Read the container size
-  unsigned int size = 0;
-  stream.read((char *)&size, sizeof(size));
+  std::size_t size = 0;
+  dataLoad(stream, size, nullptr);
+
+  auto & values = this->set();
 
   // If the current container is undersized, expand it to fit the loaded data
-  if (this->get().size() < size)
-    this->set().resize(size);
+  if (values.size() < size)
+    values.resize(size);
 
   // Load each entry of the list directly into the storage
-  for (auto & val : this->set())
+  // Because we don't shrink the container if the stored size is smaller than
```

Comment:
why don't we shrink?

Reply:
Example: user requests old and older postprocessor values in another object. This means the storage goes from size 1 to size 3. We then go to load from restart. Well, the restart only has a current value. Shrink would mean to undo the user request for the user requesting the old and older values. Instead, we want to keep the user's request for old and older, pull in current from restart, and then copy current from restart back to old and older later.